### ISSUE TITLE
Add rule to not allow plain JOIN

### ISSUE THREAD

**Issue by alexscott-ff** [sqlfluff/sqlfluff#1712]
Add rule to not allow "naked" joins, i.e. `JOIN`.

```sql
SELECT
  ...
FROM
  customer
  -- Should be an explicit join type, e.g. INNER JOIN or LEFT JOIN
  JOIN orders ON customer.id = orders.customer_id
```

Potentially, you could also add a setting to assume all naked `JOIN`s are `INNER JOIN`s (as most dialects do) and implement that when running `fix`.

```sql
SELECT
  ...
FROM
  customer
  -- Fixed
  INNER JOIN orders ON customer.id = orders.customer_id
```

**Comment by jpy-smart** [comment]
@tunetheweb could I be assigned this one please :)

**Comment by jpy-git** [comment]
@tunetheweb could I be assigned this one please :)

**Comment by tunetheweb** [comment]
Sure

We don't really use the assign label that much on this project so feel free to just leave a comment like "I'll look at this one" and then we'll consider it yours!
